src/integrationTest/java/uk/gov/hmcts/divorce/solicitor/Applicant2SolicitorSwitchToSoleFoIT.java

```diff
@@ -0,0 +1,184 @@
+import static uk.gov.hmcts.divorce.testutil.TestDataHelper.callbackRequest;
+import static uk.gov.hmcts.divorce.testutil.TestDataHelper.validJointApplicant1CaseData;
+import static uk.gov.hmcts.divorce.testutil.TestResourceUtil.expectedResponse;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+@DirtiesContext
+@ContextConfiguration(initializers = {
+    IdamWireMock.PropertiesInitializer.class})
+public class Applicant2SolicitorSwitchToSoleFoIT {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final String APPLICANT_2_SOLICITOR_SWITCH_TO_SOLE_FO_RESPONSE =
+        "classpath:solicitor-applicant2-switch-to-sole-fo-response.json";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private NotificationService notificationService;
+
+    @MockBean
+    private CcdAccessService ccdAccessService;
+
+    @MockBean
+    private IdamService idamService;
+
+    @MockBean
+    private AuthTokenGenerator serviceTokenGenerator;
+
+    @MockBean
+    private WebMvcConfig webMvcConfig;
+
+    @MockBean
+    private CaseAssignmentApi caseAssignmentApi;
+
+    @BeforeAll
+    static void setUp() {
+        OBJECT_MAPPER.registerModule(new JavaTimeModule());
+        IdamWireMock.start();
+    }
+
+    @AfterAll
+    static void tearDown() {
+        IdamWireMock.stopAndReset();
+    }
+
+    @Test
+    void shouldSwitchApplicationTypeToSoleSwitchUsersDataAndGenerateConditionalOrderAnswers()
+        throws Exception {
+
+        final String app1SolicitorEmail = "app1solicitor@test.com";
+        final String app2SolicitorEmail = "app2solicitor@test.com";
+
+        CaseData data = validJointApplicant1CaseData();
+        data.getApplication().setIssueDate(LOCAL_DATE);
+        data.getApplicant1().setSolicitorRepresented(YES);
+        data.getApplicant1().setSolicitor(
+            Solicitor.builder()
+                .name("App1 Sol")
+                .firmName("App1 Sol Firm")
+                .email(app1SolicitorEmail)
+                .build()
+        );
+        data.getApplicant2().setSolicitorRepresented(YES);
+        data.getApplicant2().setSolicitor(
+            Solicitor.builder()
+                .name("App2 Sol")
+                .firmName("App2 Sol Firm")
+                .email(app2SolicitorEmail)
+                .build()
+        );
+
+        data.setFinalOrder(
+            FinalOrder.builder()
+                .doesApplicant2WantToApplyForFinalOrder(YES)
+                .applicant2AppliedForFinalOrderFirst(YES)
+                .applicant2CanIntendToSwitchToSoleFo(YES)
+                .applicant2IntendsToSwitchToSole(Set.of(FinalOrder.IntendsToSwitchToSole.I_INTEND_TO_SWITCH_TO_SOLE))
+                .doesApplicant2IntendToSwitchToSole(YES)
+                .dateApplicant2DeclaredIntentionToSwitchToSoleFo(LocalDate.now())
```

Comment:
dateApplicant2DeclaredIntentionToSwitchToSoleFo needs to be a static date here, because in response json we are expecting 2022-11-11